XFree86-3.3/xc/programs/Xserver/hw/xfree86/XF86Conf.man
.\" $XFree86: xc/programs/Xserver/hw/xfree86/XF86Conf.man,v 3.52.2.4 1997/05/29 14:01:01 dawes Exp $
.TH XF86Config 4/5 "Version 3.3"  "XFree86"
.SH NAME
XF86Config - Configuration File for XFree86
.SH DESCRIPTION
.I XFree86
uses a configuration file called \fBXF86Config\fP for its initial
setup.  This configuration file is searched for in the following
places:
.sp
.in 8
.nf
/etc/XF86Config
<XRoot>/lib/X11/XF86Config.\fIhostname\fP
<XRoot>/lib/X11/XF86Config
.fi
.in -8
.LP
where <XRoot> refers to the root of the X11 install tree.
When an X server is started by a `root' user, it will first search for
an \fBXF86Config\fP file in that user's home directory.
.PP
This file is composed of a number of sections.  Each section has
the form:
.sp
.in 8
.nf
Section "\fISectionName\fP"
   \fISectionEntry\fP
   ...
EndSection
.fi
.in -8
.PP
The section names are:
.sp
.in 8
.nf
\fBFiles\fP (File pathnames)
\fBModule\fP (Dynamic module loading)
\fBServerFlags\fP (Server flags)
\fBKeyboard\fP (Keyboard configuration)
\fBPointer\fP (Pointer configuration)
\fBMonitor\fP (Monitor description)
\fBDevice\fP (Graphics device description)
\fBScreen\fP (Screen configuration)
\fBXInput\fP (Extended Input devices configuration)
.fi
.PP
The \fBFiles\fP section is used to specify the default font path
and the path to the RGB database.  These paths can also be set from
the command line (see \fIXserver(1)\fP).  The entries available
for this section are:
.TP 8
.B FontPath \fI"path"\fP
sets the search path for fonts.  This path is a comma separated
list of directories which the X server searches for font databases.
Multiple \fBFontPath\fP entries may be specified, and they will be
concatenated to build up the fontpath used by the server.
.sp
X11R6 allows the X server to request fonts from a font server.  A
font server is specified by placing a "<trans>/<hostname>:<port_number>"
entry into the fontpath.  For example, the fontpath
"/usr/X11R6/lib/X11/fonts/misc/,tcp/zok:7100" tells the X server
to first try to locate the font in the local directory
/usr/X11R6/lib/X11/fonts/misc.  If that fails, then request the
font from the \fIfont server\fP running on machine zok listening
for connections on TCP port number 7100.
.TP 8
.B RGBPath \fI"path"\fP
sets the path name for the RGB color database.
.TP 8
.B ModulePath \fI"path"\fP
sets the search path for dynamic modules.  This path is a comma separated
list fo directories which the X server searches for dynamic module loading
in the order specified.
Multiple \fBModulePath\fP entries may be specified, and they will be
concatenated to build the modulepath used by the server.
.PP
The
.B Module
section is used to specify which dynamic modules should be loaded.
At present dynamic modules are only used for XInput devices, and are
only supported on some systems (currently Linux ELF, FreeBSD 2.x and
NetBSD 1.x).  The entry available for this section is:
.TP 8
.B Load \fI"module"\fP
This instructs the server to load \fI"module"\fP.  If the module
is not specified with a full pathname, the directories specified in
the \fBModulePath\fP are searched.
Modules are currently available to support extended input devices.  The
names of these are:
.sp
.in 20
.nf
.B xf86Elo.so
.B xf86Jstk.so
.B xf86Wacom.so
.B xf86Summa.so
.fi
.in -20
The PEX and XIE extension are also be available as modules on some
systems. The names for these are:
.sp
.in 20
.nf
.B pex5.so
.B xie.so
.fi
.in -20
.RS 8
For an up-to-date listing, check in <XRoot>/lib/modules.
.RE
.PP
The \fBServerFlags\fP section is used to specify some miscellaneous
X server options.  The entries available for this section are:
.TP 8
.B NoTrapSignals
This prevents the X server from trapping a range of unexpected
fatal signals and exiting cleanly.  Instead, the X server will die
and drop core where the fault occurred.  The default behaviour is
for the X server exit cleanly, but still drop a core file.  In
general you never want to use this option unless you are debugging
an X server problem.
.TP 8
.B DontZap
This disallows the use of the \fBCtrl+Alt+Backspace\fP sequence.
This sequence allows you to terminate the X server.
Setting \fBDontZap\fP allows this key sequence to be passed to clients.
.TP 8
.B DontZoom
This disallows the use of the \fBCtrl+Alt+Keypad-Plus\fP and
\fBCtrl+Alt+Keypad-Minus\fP sequences.  These sequences allows you to
switch between video modes.
Setting \fBDontZoom\fP allows these key sequences to be passed to clients.
.TP 8
.B AllowNonLocalXvidtune
This allows the xvidtune client to connect from another host.  By default
non-local connections are not allowed.
.TP 8
.B DisableVidMode
This disables the parts of the VidMode extension used by the xvidtune client
that can be used to change the video modes.
.TP 8
.B AllowNonLocalModInDev
This allows a client to connect from another host and change keyboard
and mouse settings in the running server.  By default
non-local connections are not allowed.
.TP 8
.B DisableModInDev
This disables the parts of the Misc extension that can be used to
modify the input device settings dynamically.
.TP 8
.B AllowMouseOpenFail
This allows the server to start up even if the mouse device can't be
opened/initialised.
.PP
The \fBKeyboard\fP section is used to specify the keyboard input
device, parameters and some default keyboard mapping options.  The
entries available for this section are:
.TP 8
.B Protocol \fI"kbd-protocol"\fP
\fIkbd-protocol\fP may be either \fBStandard\fP or \fBXqueue\fP.
\fBXqueue\fP is specified when using the event queue driver on SVR3
or SVR4.
.TP
.B AutoRepeat \fIdelay rate\fP
changes the behavior of the autorepeat of the keyboard.  This does
not work on all platforms.
.TP 8
.B ServerNumLock
forces the X server to handle the numlock key internally.  The X
server sends a different set of keycodes for the numpad when the
numlock key is active.  This enables applications to make use of
the numpad.
.PP
.nf
.B LeftAlt \fImapping\fP
.B RightAlt \fImapping\fP
.B AltGr \fImapping\fP
.B ScrollLock \fImapping\fP
.B RightCtl \fImapping\fP
.fi
.RS 8
Allows a default mapping to be set for the above keys (note that
\fBAltGr\fP is a synonym for \fBRightAlt\fP).  The values that may
be specified for \fImapping\fP are:
.sp
.in 20
.nf
Meta
Compose
ModeShift
ModeLock
ScrollLock
Control
.fi
.PP
The default mapping when none of these options are specified is:
.sp
.in 20
.nf
LeftAlt     Meta
RightAlt    Meta
ScrollLock  Compose
RightCtl    Control
.fi
.RE
.TP 8
.B XLeds \fIled\fP ...
makes \fIled\fP available for clients instead of using the traditional
function (Scroll Lock, Caps Lock & Num Lock).  \fIled\fP is a list
of numbers in the range 1 to 3.
.TP 8
.B VTSysReq
enables the SYSV-style VT switch sequence for non-SYSV systems
which support VT switching.  This sequence is Alt-SysRq followed
by a function key (Fn).  This prevents the X server trapping the
keys used for the default VT switch sequence.
.TP 8
.B VTInit \fI"command"\fP
Runs \fIcommand\fP after the VT used by the server has been opened.
The command string is passed to "/bin/sh -c", and is run with the
real user's id with stdin and stdout set to the VT.  The purpose
of this option is to allow system dependent VT initialisation
commands to be run.  One example is a command to disable the 2-key
VT switching sequence which is the default on some systems.
.TP 8
.B XkbDisable
Turns the XKEYBOARD extension off, equivalent to using the -kb
command line option.
.PP
.nf
.B XkbRules \fI"rulesfile"\fP
.B XkbModel \fI"model"\fP
.B XkbLayout \fI"layout"\fP
.B XkbVariant \fI"variant"\fP
.B XkbOptions \fI"optionlist"\fP
.fi
.RS 8
These specify the definitions which are used to determine which
XKEYBOARD components to use.  The optionlist, should be a comma
separated list of options.
The default mappings for each these are:
.sp
.in 20
.nf
XkbRules     "xfree86"
XkbModel     "pc101"
XkbLayout    "us"
XkbVariant   ""
XkbOptions   ""
.fi
.RE
.PP
This is the preferred method of specifying the keyboard configuration,
however, you can also specify the components directly with:
.sp
.nf
.B XkbKeymap \fI"keymap"\fP
.B XkbKeycodes \fI"keycodes"\fP
.B XkbTypes \fI"types"\fP
.B XkbCompat \fI"compat"\fP
.B XkbSymbols \fI"symbols"\fP
.B XkbGeometry \fI"geometry"\fP
.fi
.RS 8
If you specify only some of the components, the remaining components
will use these default values:
.sp
.in 20
.nf
XkbKeymap    none
XkbKeycodes  "xfree86"
XkbTypes     "default"
XkbCompat    "default"
XkbSymbols   "us(pc101)"
XkbGeometry  "pc"
.fi
.RE
.PP
The \fBPointer\fP section is used to specify the pointer device
and parameters.  The entries available for this section are:
.TP 8
.B Protocol \fI"protocol-type\fP"
specifies the pointer device protocol type.  The protocol types
available are:
.sp
.in 20
.nf
.B BusMouse
.B Logitech
.B Microsoft
.B MMSeries
.B Mouseman
.B MouseSystems
.B PS/2
.B MMHitTab
.B GlidePoint
.B IntelliMouse
.B Xqueue
.B OSMouse
.fi
.in -20
.RS 8
.PP
One should specify \fBBusMouse\fP for the Logitech bus mouse.
Also, many newer Logitech serial mice use either the \fBMicrosoft\fP
or \fBMouseMan\fP protocol.  \fBXqueue\fP should be specified here
if it was used in the \fBKeyboard\fP section.  \fBOSMouse\fP refers
to the event-driver mouse interface available on SCO's SVR3, and the
mouse interface provided for OS/2.  This
may optionally be followed by a number specifying the number of
buttons the mouse has.
.PP
If you have a mouse connected to a PS/2 port, you should specify \fBPS/2\fP,
regardless of the type of mouse you are using.
.RE
.TP 8
.B Device \fI"pointer-dev"\fP
specifies the device the server should open for pointer input (eg,
\fB/dev/tty00\fP or \fB/dev/mouse\fP).  A device should not be
specified when using the \fBXqueue\fP or \fBOSMouse\fP protocols.
.TP 8
.B Port \fI"pointer-dev"\fP
is an alternate form of the \fBDevice\fP entry.
.TP 8
.B BaudRate \fIrate\fP
sets the baudrate of the serial mouse to \fIrate\fP. For mice that
allow dynamic speed adjustments (like older Logitechs) the baudrate is
changed in the mouse.  Otherwise the rate is simply set on the
computer's side to allow mice with non-standard rates (the standard
rate is 1200).  For 99% of mice you should not set this to anything
other than the default (1200).
.TP 8
.B Emulate3Buttons
enables the emulation of the third mouse button for mice which only
have two physical buttons.  The third button is emulated by pressing
both buttons simultaneously.
.TP 8
.B Emulate3Timeout \fItimeout\fP
sets the time (in milliseconds) that the server waits before deciding if
two buttons were pressed ``simultaneously'' when 3 button emulation is
enabled.  The default timeout is 50ms.
.TP 8
.B ChordMiddle
handles mice which send left+right events when the middle button
is used (like some Logitech Mouseman mice).
.TP 8
.B SampleRate \fIrate\fP
sets the number of motion/button-events the mouse sends per second.
This is currently only supported for some Logitech mice.
.TP 8
.B ClearDTR
This option clears the DTR line on the serial port used by the
mouse.  This option is only valid for a mouse using the \fBMouseSystems\fP
protocol.  Some dual-protocol mice require DTR to be cleared to
operate in MouseSystems mode.  Note, in versions of XFree86 prior
to 2.1, this option also cleared the RTS line.  A separate
\fBClearRTS\fP option has since been added for mice which require
this.
.TP 8
.B ClearRTS
This option clears the RTS line on the serial port used by the
mouse.  This option is only valid for a mouse using the \fBMouseSystems\fP
protocol.  Some dual-protocol mice require both DTR and RTS to be
cleared to operate in MouseSystems mode.  Both the \fBClearDTR\fP
and \fBClearRTS\fP options should be used for such mice.
.PP
The \fBMonitor\fP sections are used to define the specifications
of a monitor and a list of video modes suitable for use with a
monitor.  More than one \fBMonitor\fP section may be present in an

ace/ChangeLog
Fri Oct 05 05:56:50 2001  Balachandran  <bala@cs.wustl.edu>

        * ACE version 5.2 released.

Thu Oct  4 17:38:21 2001  Steve Huston  <shuston@riverace.com>

        * ace/Synch.h: Added some comments to explain the rationale for
          the different return values from the blocking and timed
          versions of ACE_Null_Semaphore::acquire().  Thanks to Doug
          Schmidt for explaining this.

Thu Oct  4 17:27:39 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Handler.cpp (recv_log_record): Insure that
          the payload message block is properly aligned for any CDR
          extraction. Many thanks to Bala for noticing this problem and
          sending code for the fix.

Thu Oct  4 17:08:39 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Process_Per_Connection_Logging_Server.cpp:
          In prepare(), call ACE_Process_Options::command_line() rather
          than ACE_Process_Options::process_name() to set the program
          path for the child process.

Thu Oct  4 13:18:37 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Handler.cpp: Use a ACE_Message_Block,
          rather than hijacking a CDR block, to read the payload into.

Thu Oct  4 10:14:34 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Added Donald Acton to the Hall of Fame.

Thu Oct  4 06:39:20 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/config-sunos5.5.h: Added ACE_LACKS_MKSTEMP as SunOS 5.5 does
	  not support mkstemp (). 
	
	* ace/config-sunos5.6.h: Undefined ACE_LACKS_MKSTEMP as Sun 5.6
	  and above support it. 

	  Thanks to Don Hinton <dhinton@swan.fr> for reporting this
	  problem and Chad Elliott for testing this out on a Sun OS 5.5
	  box. 

Wed Oct  3 16:37:41 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/CDR_Stream.cpp (read_1): Reverted this change "Wed Oct  3
	  15:05:43 2001  Jeff Parsons <parsons@cs.wustl.edu>".  This
	  change is alone not enough to fix the problem that Brad
	  reported. The sample program for the text needs to be fixed to
	  get it working right.

Wed Oct  3 15:05:43 2001  Jeff Parsons <parsons@cs.wustl.edu>

        * ace/CDR_Stream.cpp (read_1);

          Reverted changed made in 

          Fri Aug 17 16:56:16 2001  Jeff Parsons <parsons@cs.wustl.edu>

          to read_1(). A similar change made in several other methods
          is ok, but in this case bytes can be copied into the
          input CDR stream without changing wr_ptr, so the check will
          fail unnecessarily. Thanks to Brad Hoskins <BHoskins@slo.newport.com>
          for pointing this out.

Wed Oct  2 12:18:37 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Server.cpp: Fixed spelling error.
        * examples/C++NPv1/Logging_Handler.cpp: Comment length adjustment.

Tue Oct  2 21:52:31 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Client.cpp: Reverted this:
          Mon Oct  1 18:27:41 2001  Steve Huston  <shuston@riverace.com>

Tue Oct  2 19:52:57 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* Makefile: Removed some of the files that no longer exist in the
	  repository. The inclusion of these files in the Makefile added
	  noise during the beta cutting process.

Tue Oct 02 15:34:49 2001  Carlos O'Ryan  <coryan@uci.edu>

	* etc/tao_cosevent.doxygen:
	* etc/tao_cosnotification.doxygen:
	  Fixed typos in the tag file names.

Tue Oct  2 15:52:27 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Added Sunny Leung to the Hall of Fame.

Tue Oct  2 15:33:36 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Removed one instance of Peter Phillips
	  <pphillip@opentext.com> from the Hall of Fame.

Tue Oct  2 15:00:37 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/CDR_Stream.cpp (clone_from): Copied the GIOP major and minor
	  versions from the incoming CDR stream. Thanks to Sunny Leung 
	  <sleung@opentext.com> for reporting this. This closes bugzilla
	  #1043.  

Tue Oct  2 09:45:39 2001  Steve Huston  <shuston@riverace.com>

        * tests/libDLL_Test.icc:
        * tests/libService_Config_DLL.icc: Add link(exportAll) to the link
          options to export the entrypoints needed for the tests.

Mon Oct  1 20:04:14 2001  Priyanka Gontla  <pgontla@ece.uci.edu>

	* THANKS: 
	  Added Tushar Nair to the THANKS file.

Mon Oct  1 18:06:48 2001  Priyanka Gontla  <pgontla@ece.uci.edu>

	* ace/Object_Manager.cpp (ACE_UnhandledExceptionFilter):
	  Removed the un-needed #if 0's. Left them before as a safety
	  measure (in case something doesnt work) and forgot to remove
	  them later.

	* ace/Log_Msg.cpp (instance):
	  Fixed the seg faults that are raised when ace is compiled with
	  ACE_NTRACE=0 on SunOS. Added a check before locking and
	  releasing a lock.

	  Thanks to Tushar Nair <ntushar@yahoo.com> for reporting the
	  problem and for providing a fix.

Mon Oct  1 18:27:41 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Client.cpp: Remove the third (optional)
          arg from getline(). Using a char literal without further
          qualification causes confusion in Unicode builds.

Mon Oct  1 14:44:13 2001  Ossama Othman  <ossama@uci.edu>

	* ace/SSL/SSL_Context.cpp (ssl_library_init):

	  Initialize the locking callbacks in the crypto library before
	  initializing the OpenSSL library.  OpenSSL may require the
	  locking callbacks to be initialized before its own
	  initialization.

	* ace/SSL/SSL_Context.cpp (ssl_library_fini):

	  Perform crypto library resource cleanup before reclaiming the
	  locks used by the locking callbacks.  Crypto library resource
	  cleanup makes use of the locking callbacks.


XFree86-3.3/xc/programs/Xserver/hw/xfree86/XF86Conf.cpp
XCOMM $XFree86: xc/programs/Xserver/hw/xfree86/XF86Conf.cpp,v 3.29 1996/12/23 06:30:30 dawes Exp $
XCOMM
XCOMM Copyright (c) 1994 by The XFree86 Project, Inc.
XCOMM
XCOMM Permission is hereby granted, free of charge, to any person obtaining a
XCOMM copy of this software and associated documentation files (the "Software"),
XCOMM to deal in the Software without restriction, including without limitation
XCOMM the rights to use, copy, modify, merge, publish, distribute, sublicense,
XCOMM and/or sell copies of the Software, and to permit persons to whom the
XCOMM Software is furnished to do so, subject to the following conditions:
XCOMM 
XCOMM The above copyright notice and this permission notice shall be included in
XCOMM all copies or substantial portions of the Software.
XCOMM 
XCOMM THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
XCOMM IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
XCOMM FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
XCOMM THE XFREE86 PROJECT BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
XCOMM WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF
XCOMM OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
XCOMM SOFTWARE.
XCOMM 
XCOMM Except as contained in this notice, the name of the XFree86 Project shall
XCOMM not be used in advertising or otherwise to promote the sale, use or other
XCOMM dealings in this Software without prior written authorization from the
XCOMM XFree86 Project.
XCOMM
XCOMM $XConsortium: XF86Conf.cpp /main/22 1996/10/23 11:43:51 kaleb $

XCOMM **********************************************************************
XCOMM Refer to the XF86Config(4/5) man page for details about the format of 
XCOMM this file. This man page is installed as MANPAGE 
XCOMM **********************************************************************

XCOMM **********************************************************************
XCOMM Files section.  This allows default font and rgb paths to be set
XCOMM **********************************************************************

Section "Files"

XCOMM The location of the RGB database.  Note, this is the name of the
XCOMM file minus the extension (like ".txt" or ".db").  There is normally
XCOMM no need to change the default.

    RgbPath	RGBPATH

XCOMM Multiple FontPath entries are allowed (which are concatenated together),
XCOMM as well as specifying multiple comma-separated entries in one FontPath
XCOMM command (or a combination of both methods)

    FontPath	MISCFONTPATH
USE_75FONTS	DPI75USFONTPATH
USE_100FONTS	DPI100USFONTPATH
USE_T1FONTS	T1FONTPATH
USE_SPFONTS	SPFONTPATH
USE_75FONTS	DPI75FONTPATH
USE_100FONTS	DPI100FONTPATH

XCOMM For OSs that support Dynamically loaded modules, ModulePath can be
XCOMM used to set a search path for the modules.  This is currently supported
XCOMM for Linux ELF, FreeBSD 2.x and NetBSD 1.x.  The default path is shown
XCOMM here.

XCOMM    ModulePath	MODULEPATH

EndSection

XCOMM **********************************************************************
XCOMM Module section -- this is an optional section which is used to specify
XCOMM which dynamically loadable modules to load.  Dynamically loadable
XCOMM modules are currently supported only for Linux ELF, FreeBSD 2.x
XCOMM and NetBSD 1.x.  Currently, dynamically loadable modules are used
XCOMM only for some extended input (XInput) device drivers.
XCOMM **********************************************************************
XCOMM
XCOMM Section "Module"
XCOMM
XCOMM This loads the module for the Joystick driver
XCOMM
XCOMM Load "xf86Jstk.so"
XCOMM
XCOMM EndSection

XCOMM **********************************************************************
XCOMM Server flags section.
XCOMM **********************************************************************

Section "ServerFlags"

XCOMM Uncomment this to cause a core dump at the spot where a signal is 
XCOMM received.  This may leave the console in an unusable state, but may
XCOMM provide a better stack trace in the core dump to aid in debugging

XCOMM    NoTrapSignals

XCOMM Uncomment this to disable the <Crtl><Alt><BS> server abort sequence
XCOMM This allows clients to receive this key event.

XCOMM    DontZap

XCOMM Uncomment this to disable the <Crtl><Alt><KP_+>/<KP_-> mode switching
XCOMM sequences.  This allows clients to receive these key events.

XCOMM    DontZoom

XCOMM Uncomment this to disable tuning with the xvidtune client. With
XCOMM it the client can still run and fetch card and monitor attributes,
XCOMM but it will not be allowed to change them. If it tries it will
XCOMM receive a protocol error.

XCOMM	DisableVidModeExtension

XCOMM Uncomment this to enable the use of a non-local xvidtune client.

XCOMM	AllowNonLocalXvidtune

XCOMM Uncomment this to disable dynamically modifying the input device
XCOMM (mouse and keyboard) settings.

XCOMM	DisableModInDev

XCOMM Uncomment this to enable the use of a non-local client to
XCOMM change the keyboard or mouse settings (currently only xset).

XCOMM	AllowNonLocalModInDev

EndSection

XCOMM **********************************************************************
XCOMM Input devices
XCOMM **********************************************************************

XCOMM **********************************************************************
XCOMM Keyboard section
XCOMM **********************************************************************

Section "Keyboard"

    Protocol	"Standard"

XCOMM when using XQUEUE, comment out the above line, and uncomment the
XCOMM following line

XCOMM    Protocol	"Xqueue"

    AutoRepeat	500 5

XCOMM Let the server do the NumLock processing.  This should only be required
XCOMM when using pre-R6 clients
XCOMM    ServerNumLock

XCOMM Specifiy which keyboard LEDs can be user-controlled (eg, with xset(1))
XCOMM    Xleds      1 2 3

XCOMM To set the LeftAlt to Meta, RightAlt key to ModeShift, 
XCOMM RightCtl key to Compose, and ScrollLock key to ModeLock:

XCOMM    LeftAlt     Meta
XCOMM    RightAlt    ModeShift
XCOMM    RightCtl    Compose
XCOMM    ScrollLock  ModeLock

XCOMM To disable the XKEYBOARD extension, uncomment XkbDisable.

XCOMM XkbDisable

XCOMM To customise the XKB settings to suit your keyboard, modify the
XCOMM lines below (which are the defaults).  For example, for a non-U.S.
XCOMM keyboard, you will probably want to use:
XCOMM    XkbModel    "pc102"
XCOMM If you have a Microsoft Natural keyboard, you can use:
XCOMM    XkbModel    "microsoft"
XCOMM
XCOMM Then to change the language, change the Layout setting.
XCOMM For example, a german layout can be obtained with:
XCOMM    XkbLayout   "de"
XCOMM or:
XCOMM    XkbLayout   "de"
XCOMM    XkbVariant  "nodeadkeys"
XCOMM
XCOMM If you'd like to switch the positions of your capslock and
XCOMM control keys, use:
XCOMM    XkbOptions  "ctrl:swapcaps"


XCOMM These are the default XKB settings for XFree86
XCOMM    XkbRules    "xfree86"
XCOMM    XkbModel    "pc101"
XCOMM    XkbLayout   "us"
XCOMM    XkbVariant  ""
XCOMM    XkbOptions  ""

EndSection


XCOMM **********************************************************************
XCOMM Pointer section
XCOMM **********************************************************************

Section "Pointer"

    Protocol	"Microsoft"
    Device	MOUSEDEV

XCOMM When using XQUEUE, comment out the above two lines, and uncomment
XCOMM the following line.

XCOMM    Protocol	"Xqueue"

XCOMM Baudrate and SampleRate are only for some Logitech mice

XCOMM    BaudRate	9600
XCOMM    SampleRate	150

XCOMM Emulate3Buttons is an option for 2-button Microsoft mice
XCOMM Emulate3Timeout is the timeout in milliseconds (default is 50ms)

XCOMM    Emulate3Buttons
XCOMM    Emulate3Timeout	50

XCOMM ChordMiddle is an option for some 3-button Logitech mice

XCOMM    ChordMiddle

EndSection


XCOMM **********************************************************************
XCOMM Xinput section -- this is optional and is required only if you
XCOMM are using extended input devices.  This is for example only.  Refer
XCOMM to the XF86Config man page for a description of the options.
XCOMM **********************************************************************
XCOMM
XCOMM Section "Xinput"
XCOMM    SubSection "WacomStylus"
XCOMM        Port "/dev/ttyS1"
XCOMM        DeviceName "Wacom"
XCOMM    EndSubSection
XCOMM    SubSection "WacomCursor"
XCOMM        Port "/dev/ttyS1"
XCOMM    EndSubSection
XCOMM    SubSection "WacomEraser"
XCOMM        Port "/dev/ttyS1"
XCOMM    EndSubSection
XCOMM
XCOMM    SubSection "Elographics"
XCOMM        Port "/dev/ttyS1"
XCOMM        DeviceName "Elo"
XCOMM        MinimumXPosition 300
XCOMM        MaximumXPosition 3500
XCOMM        MinimumYPosition 300
XCOMM        MaximumYPosition 3500
XCOMM        Screen 0
XCOMM        UntouchDelay 10
XCOMM        ReportDelay 10
XCOMM    EndSubSection
XCOMM   
XCOMM    SubSection "Joystick"
XCOMM        Port "/dev/joy0"
XCOMM        DeviceName "Joystick"
XCOMM        TimeOut 10
XCOMM        MinimumXPosition 100
XCOMM        MaximumXPosition 1300
XCOMM        MinimumYPosition 100
XCOMM        MaximumYPosition 1100
XCOMM        # CenterX 700
XCOMM        # CenterY 600
XCOMM        Delta 20
XCOMM    EndSubSection
XCOMM
XCOMM The Mouse Subsection contains the same type of entries as the
XCOMM standard Pointer Section (see above), with the addition of the
XCOMM DeviceName entry.
XCOMM
XCOMM    SubSection "Mouse"
XCOMM        Port "/dev/mouse2"
XCOMM        DeviceName "Second Mouse"
XCOMM        Protocol "Logitech"
XCOMM    EndSubSection
XCOMM EndSection


XCOMM **********************************************************************
XCOMM Monitor section
XCOMM **********************************************************************

XCOMM Any number of monitor sections may be present

Section "Monitor"

    Identifier	"Generic Monitor"
    VendorName	"Unknown"
    ModelName	"Unknown"

XCOMM HorizSync is in kHz unless units are specified.
XCOMM HorizSync may be a comma separated list of discrete values, or a
XCOMM comma separated list of ranges of values.
XCOMM NOTE: THE VALUES HERE ARE EXAMPLES ONLY.  REFER TO YOUR MONITOR'S
XCOMM USER MANUAL FOR THE CORRECT NUMBERS.

    HorizSync   31.5  # typical for a single frequency fixed-sync monitor

XCOMM    HorizSync	30-64         # multisync
XCOMM    HorizSync	31.5, 35.2    # multiple fixed sync frequencies
XCOMM    HorizSync	15-25, 30-50  # multiple ranges of sync frequencies

XCOMM VertRefresh is in Hz unless units are specified.
XCOMM VertRefresh may be a comma separated list of discrete values, or a
XCOMM comma separated list of ranges of values.
XCOMM NOTE: THE VALUES HERE ARE EXAMPLES ONLY.  REFER TO YOUR MONITOR'S
XCOMM USER MANUAL FOR THE CORRECT NUMBERS.

    VertRefresh 60  # typical for a single frequency fixed-sync monitor

XCOMM    VertRefresh	50-100        # multisync
XCOMM    VertRefresh	60, 65        # multiple fixed sync frequencies
XCOMM    VertRefresh	40-50, 80-100 # multiple ranges of sync frequencies

XCOMM Modes can be specified in two formats.  A compact one-line format, or
XCOMM a multi-line format.

XCOMM A generic VGA 640x480 mode (hsync = 31.5kHz, refresh = 60Hz)
XCOMM These two are equivalent

XCOMM    ModeLine "640x480" 25.175 640 664 760 800 480 491 493 525

    Mode "640x480"
        DotClock	25.175
        HTimings	640 664 760 800
        VTimings	480 491 493 525
    EndMode

XCOMM These two are equivalent

XCOMM    ModeLine "1024x768i" 45 1024 1048 1208 1264 768 776 784 817 Interlace

XCOMM    Mode "1024x768i"
XCOMM        DotClock	45
XCOMM        HTimings	1024 1048 1208 1264
XCOMM        VTimings	768 776 784 817
XCOMM        Flags		"Interlace"
XCOMM    EndMode

EndSection

XCOMM **********************************************************************
XCOMM Graphics device section
XCOMM **********************************************************************

XCOMM Any number of graphics device sections may be present

Section "Device"
    Identifier	"Generic VGA"
    VendorName	"Unknown"
    BoardName	"Unknown"
    Chipset	"generic"
XCOMM    VideoRam	256
XCOMM    Clocks	25.2 28.3
EndSection

Section "Device"
    # SVGA server auto-detected chipset
    Identifier	"Generic SVGA"
    VendorName	"Unknown"
    BoardName	"Unknown"
EndSection

XCOMM Section "Device"
XCOMM    Identifier	"Any Trident TVGA 9000"
XCOMM    VendorName	"Trident"
XCOMM    BoardName	"TVGA 9000"
XCOMM    Chipset	"tvga9000"
XCOMM    VideoRam	512
XCOMM    Clocks	25 28 45 36 57 65 50 40 25 28 0 45 72 77 80 75
XCOMM EndSection

XCOMM Section "Device"
XCOMM    Identifier	"Actix GE32+ 2MB"
XCOMM    VendorName	"Actix"
XCOMM    BoardName	"GE32+"
XCOMM    Ramdac	"ATT20C490"
XCOMM    Dacspeed	110
XCOMM    Option	"dac_8_bit"
XCOMM    Clocks	 25.0  28.0  40.0   0.0  50.0  77.0  36.0  45.0
XCOMM    Clocks	130.0 120.0  80.0  31.0 110.0  65.0  75.0  94.0
XCOMM EndSection

XCOMM Section "Device"
XCOMM    Identifier	"Hercules mono"
XCOMM EndSection

XCOMM **********************************************************************
XCOMM Screen sections
XCOMM **********************************************************************

XCOMM The colour SVGA server

Section "Screen"
    Driver	"svga"
    Device	"Generic SVGA"
    Monitor	"Generic Monitor"
    DefaultColorDepth 8
    Subsection "Display"
        Depth	    8
        Modes	    "640x480"
        ViewPort    0 0
        Virtual     800 600
    EndSubsection
EndSection

XCOMM The 16-colour VGA server

Section "Screen"
    Driver	"vga16"
    Device	"Generic VGA"
    Monitor	"Generic Monitor"
    Subsection "Display"
        Modes	    "640x480"
        ViewPort    0 0
        Virtual     800 600
    EndSubsection
EndSection

XCOMM The Mono server

Section "Screen"
    Driver	"vga2"
    Device	"Generic VGA"
    Monitor	"Generic Monitor"
    Subsection "Display"
        Modes	    "640x480"
        ViewPort    0 0
        Virtual     800 600
    EndSubsection
EndSection

XCOMM The hercules driver in the Mono and VGA16 servers
XCOMM
XCOMM Section "Screen"
XCOMM     Driver	"mono"
XCOMM     Device	"Hercules Mono"
XCOMM     Monitor	"Generic Monitor"
XCOMM     Subsection "Display"
XCOMM     EndSubsection
XCOMM EndSection

XCOMM The accelerated servers (S3, Mach32, Mach8, 8514, P9000, AGX, W32)

XCOMM Section "Screen"
XCOMM     Driver	"accel"
XCOMM     Device	"Actix GE32+ 2MB"
XCOMM     Monitor	"Generic Monitor"
XCOMM     DefaultColorDepth 8
XCOMM     Subsection  "Display"
XCOMM         Depth	    8
XCOMM         Modes	    "640x480"
XCOMM         ViewPort    0 0
XCOMM         Virtual	    1280 1024
XCOMM     EndSubsection
XCOMM     SubSection "Display"
XCOMM         Depth	    16
XCOMM         Weight	    565
XCOMM         Modes	    "640x480"
XCOMM         ViewPort    0 0
XCOMM         Virtual	    1024 768
XCOMM     EndSubsection
XCOMM EndSection

